core/unix/os.c

```diff
@@ -7416,6 +7422,52 @@ pre_system_call(dcontext_t *dcontext)
                                      dcontext->sys_num);
         break;
     }
+#    ifdef LINUX
+    case SYS_ppoll: {
+        kernel_sigset_t *mask = (kernel_sigset_t *)sys_param(dcontext, 3);
+        size_t sizemask = (size_t)sys_param(dcontext, 4);
+        kernel_sigset_t nullsigmask;
+        memset(&nullsigmask, 0, sizeof(nullsigmask));
+        /* original app's sigmask parameter is now NULL effectively making the syscall
+         * a non p* version, and the mask's semantics are emulated by DR instead. */
+        set_syscall_param(dcontext, 3, *(reg_t *)&nullsigmask);
```

Comment:
Ok the point was indeed to - from the app's point of view - change pselect(..., mask) into a pselect(..., NULL) call. Passing in *(reg_t *)NULL will do the same as above, what am I missing?

Reply:
I read this as passing &nullsigmask: I missed the first `*`.  So your code is passing the first 4 or 8 bytes of nullsigmask as the pointer: so scratch the "unblock all signals", but that is a strange way to try to pass a NULL pointer, right?  kernel_sigset_t is a struct: what if it's larger than a pointer, which it most certainly is for 32-bit?  Why zeroing out a struct and copy the first N bytes of it, instead of just passing NULL?

Reply:
That's right, done

Reply:
I realized what I thought when I was doing this now, but you're right. Fixed to (reg_t)NULL